llmfoundry/utils/exceptions.py

```diff
@@ -354,3 +356,26 @@ class RunTimeoutError(InternalError):
     def __init__(self, timeout: int) -> None:
         message = f'Run timed out after {timeout} seconds.'
         super().__init__(message, timeout=timeout)
+
+
+class DatasetMissingFileError(UserError):
```

Comment:
Is this error supposed to cover both cases where a shard file is missing AND the index.json metadata file is missing? From the PR description, it sounds like this error is only for a missing/badly specified index.json file. So maybe it's more appropriate to call this `DatasetMissingIndexError` instead